test/libsolidity/semanticTests/array/concat/bytes_concat_empty_strings.sol

```diff
@@ -0,0 +1,27 @@
+contract C {
+    function f() public returns (bytes memory) {
+        bytes memory b = "";
+        return bytes.concat(
+            bytes.concat(b),
+            bytes.concat(b, b),
+            bytes.concat("", b),
+            bytes.concat(b, "")
+        );
+    }
+
+    function g() public returns (bytes memory) {
+        return bytes.concat("", "abc", "");
```

Comment:
Why not include the `hex""` and `unicode""` cases in the semantic tests too?

Reply:
Sure. Looking at the implementation I think it's unlikely that they behave any differently which is why I only covered them in syntax tests but I don't mind having them here either.

Reply:
Done. Changed to `bytes.concat("", "abc", hex"", "abc", unicode"")`.